dvc/remote/base.py

```diff
@@ -891,7 +899,10 @@ def get_files_number(self, checksum):
             return 0
 
         if self.is_dir_checksum(checksum):
-            return len(self.get_dir_cache(checksum))
+            try:
+                return len(self.get_dir_cache(checksum))
+            except DirCacheError:
```

Comment:
Hm, how about we catch this in `get_dir_cache` itself and return []?

Reply:
That meems more like it was before and also wouldn't break get_dir_cache invocations outside of get_files_number